go/vt/vtgate/evalengine/is_expr.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package evalengine
+
+import (
+	"vitess.io/vitess/go/mysql/collations"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+)
+
+// IsExpr represents the IS expression in MySQL.
+// boolean_primary IS [NOT] {TRUE | FALSE | NULL}
+type IsExpr struct {
+	Expr Expr
```

Comment:
If you embed `Unary` in this struct instead of `Expr`, you'll get a lot of the methods you've implemented for free.